client/ayon_maya/api/lib.py

```diff
@@ -4579,3 +4592,89 @@ def get_scene_units_settings(project_settings=None)-> tuple[str, str]:
     linear_unit = scene_units.get("linear_units", "cm")
     angular_unit = scene_units.get("angular_units", "deg")
     return linear_unit, angular_unit
+
+
+def apply_connections(connections, target_nodes_by_id, source_nodes_by_id):
+    """Connect texture reference object nodes to the target object
+    nodes if there is one.
+
+    Args:
+        connections (list[dict[str, Any]]): list of texture reference
+        objects connection data
+        target_nodes_by_id (dict[str, list[str]]): The dict with target shape node ids
+        source_nodes_by_id (dict[str, list[str]]): The dict with texture reference node ids.
+    """
+    # Compare loaded connections to scene.
+    for connection in connections:
+        source_node = next(
+            iter(source_nodes_by_id.get(
+            connection["sourceID"], [])
+            ), None
+        )
+
+        target_nodes = target_nodes_by_id.get(connection["destinationID"], [])
+
+        if not source_node or not target_nodes:
+            self.log.debug(
+                "Could not find nodes for reference input:\n" +
+                json.dumps(connection, indent=4, sort_keys=True)
+            )
+            continue
+
+        # Skip if source node is one of our target shapes
+        if source_node in target_nodes:
+            log.debug("Skipping connection to source target: %s", connection)
+            continue
```

Comment:
It *may* solve the issue however:
- Target nodes is a `list` the lookup will be slow
- You are now excluding all the connections, not just the connection to itself... as such it'll essentially end up failing to do anything.